File: server/php/src/Application/Interfaces/SDKClients/MandateClientInterface.php
<?php

namespace OnlinePayments\ExampleApp\Application\Interfaces\SDKClients;

use OnlinePayments\ExampleApp\Application\Domain\Payments\Mandate;
use OnlinePayments\ExampleApp\Application\DTOs\Mandate\GetMandateResponseDto;
use OnlinePayments\ExampleApp\Application\DTOs\Payment\RequestDto;

interface MandateClientInterface
{
    public function createMandate(RequestDto $requestDto): Mandate;
    public function getMandate(?string $existingUniqueMandateReference): ?GetMandateResponseDto;
}

File: server/java/src/main/java/com/onlinepayments/example/service/interfaces/clients/MandateClientInterface.java
package com.onlinepayments.example.service.interfaces.clients;

import com.onlinepayments.example.service.domain.payments.Mandate;
import com.onlinepayments.example.service.dtos.mandates.CreateMandateRequestDto;
import com.onlinepayments.example.service.dtos.mandates.GetMandateResponseDto;

public interface MandateClientInterface {
    Mandate createMandate(CreateMandateRequestDto request);

    GetMandateResponseDto getMandate(String existingUniqueMandateReference);
}
